core/src/calleeFunctions/UniswapV3Callee.ts

```diff
@@ -34,7 +42,18 @@ const getMarketPrice = async function (
     collateralAmount: BigNumber
 ): Promise<BigNumber> {
     // convert collateral into DAI
-    const daiAmount = await convertCollateralToDaiUsingRoute(network, collateral.symbol, marketId, collateralAmount);
+    const { route, fees } = await getRouteAndGasQuote(network, collateral.symbol, collateralAmount, marketId);
+    if (!route) {
+        throw new Error(`No route found for ${collateral.symbol} to DAI`);
+    }
+    const pools = await routeToPool(network, route, collateral.symbol, fees);
+    const daiAmount = await convertCollateralToDaiUsingPool(
+        network,
+        collateral.symbol,
+        marketId,
+        collateralAmount,
+        pools
+    );
```

Comment:
Maybe a better explanation would've been: previously we expected that pools (/routes) are static. Now we should receive and store pools together with prices (because re-requesting price via auto-router can change the pool) – so if the pool is generated, it have to be stored in the auction together with the price (unless the price is not an auction price, but just a reference price)